homeassistant/components/qbus/switch.py

```diff
@@ -0,0 +1,91 @@
+"""Support for Qbus switch."""
+
+from typing import Any
+
+from qbusmqttapi.discovery import QbusMqttOutput
+from qbusmqttapi.state import QbusMqttOnOffState, StateType
+
+from homeassistant.components.mqtt import ReceiveMessage
+from homeassistant.components.switch import (
+    ENTITY_ID_FORMAT,
+    SwitchDeviceClass,
+    SwitchEntity,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .coordinator import QbusConfigEntry
+from .entity import QbusEntity
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: QbusConfigEntry, add_entities: AddEntitiesCallback
+) -> None:
+    """Set up switch entities."""
+    coordinator = entry.runtime_data
+
+    # Keep a record of added entities
+    known_items: list[QbusMqttOutput] = []
+
+    # Local function that calls add_entities for new entities
+    def _check_items() -> None:
+        new_items = [
+            item
+            for item in coordinator.data
+            if item.type == "onoff" and item.id not in {k.id for k in known_items}
```

Comment:
Can we create a `known_item_ids` to avoid doing `{k.id for k in known_items}` on every iteration?